Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs

```diff
@@ -13,24 +10,56 @@ namespace Terminal.Gui;
 ///     <see cref="WindowsDriver"/> - <see cref="NetDriver"/> that uses the .NET Console API - <see cref="FakeConsole"/>
 ///     for unit testing.
 /// </remarks>
-public abstract class ConsoleDriver
+public abstract class ConsoleDriver : IConsoleDriver
 {
+    /// <summary>
+    ///     Set this to true in any unit tests that attempt to test drivers other than FakeDriver.
+    ///     <code>
+    ///  public ColorTests ()
+    ///  {
+    ///    ConsoleDriver.RunningUnitTests = true;
+    ///  }
+    /// </code>
+    /// </summary>
+    internal static bool RunningUnitTests { get; set; }
+
+    /// <summary>Get the operating system clipboard.</summary>
+    public IClipboard? Clipboard { get; internal set; }
+
+    /// <summary>Returns the name of the driver and relevant library version information.</summary>
+    /// <returns></returns>
+    public virtual string GetVersionInfo () { return GetType ().Name; }
+
+    #region ANSI Esc Sequence Handling
+
+    // QUESTION: This appears to be an API to help in debugging. It's only implemented in CursesDriver and WindowsDriver.
+    // QUESTION: Can it be factored such that it does not contaminate the ConsoleDriver API?
+    /// <summary>
+    ///     Provide proper writing to send escape sequence recognized by the <see cref="ConsoleDriver"/>.
+    /// </summary>
+    /// <param name="ansi"></param>
+    public abstract void WriteRaw (string ansi);
+
+    #endregion ANSI Esc Sequence Handling
+
+    #region Screen and Contents
+
     // As performance is a concern, we keep track of the dirty lines and only refresh those.
     // This is in addition to the dirty flag on each cell.
     internal bool []? _dirtyLines;
 
     // QUESTION: When non-full screen apps are supported, will this represent the app size, or will that be in Application?
     /// <summary>Gets the location and size of the terminal screen.</summary>
-    internal Rectangle Screen => new (0, 0, Cols, Rows);
+    public Rectangle Screen => new (0, 0, Cols, Rows);
```

Comment:
I do not think this should be `public`. Why did you change it?

Reply:
This is because i added IConsoleDriver

With a public interface you cannot have internal/private members when implementing.

Question is:
- Are end users allowed to write new drivers?
- Do all drivers need to have the concept of a Clip 

If yes to both then it must be public.

If Clip is used only within the ConsoleDriver class (not even Application) then it can be removed from interface.

Reply:
This is a first pass at IConsoleDriver, I am trying to cut it down in scale as part of v2 driver work - but need to start somewhere.

This pr has already diverged a lot from my v2 branch which came off ansi-parser.

Putting interface in both branches is attempt to make merge job easier.

Already I think I will have to throw away my parser integration with these existing drivers because it is too hard to merge with the structural changes

Reply:
I do not forsee anyone outside of maintainers building another driver. In the 6+ years I've been working on this project no one has even suggested doing so. So no.

I'm ok with the interface being public, but we need to ensure `Application/View` have all the APIs needed so end-devs don't keep using driver fn directly.

Reply:
Agreed, I envisage this interface shrinking and maybe even being split into sub interfaces later but it is a journey and has to start with small steps.

As soon as a member on interface goes then associated return types eg MainLoop can go back to internal

Reply:
> I do not forsee anyone outside of maintainers building another driver. In the 6+ years I've been working on this project no one has even suggested doing so. So no.

There was  https://github.com/Blazor-Console/HACC

Also can think of lots of potential use cases 

I want to subclass ConsoleDriver but override rendering so it goes out to a file and input from a file - that way I can do automated testing.

I want to Moq console driver so I can test my new views are behaving as expected

I want to subclass ConsoleDriver but override how colors are picked so to avoid combinations that are difficult for colorblind users.

Now maybe these use cases are not tied to ConsoleDriver in the end but to IRender or IInput or something. But only way we get there is incrementally.

A big issue now is how tied Application is to ConsoleDriver and how static everything in Application is - this is next on my explore list.

Reply:
You're on the right track. My concerns are secondary. Keep going!